### PR TITLE
Added new environment value 'inmemory' that enables the in memory dat…

### PR DESCRIPTION
**Pull request by michaelkleinhenz**
Enables a new ENV value 'inmemory' that enables the in memory datasource. No files must be changed to enable inmemory anymore. Default is 'development' as before.

---
This change is [Reviewable](https://reviewable.io/reviews/almighty/almighty-ui/65)

### CODE REVIEW

File: src/app/in-memory-data.service.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { InMemoryDbService } from 'angular2-in-memory-web-api';
 export class InMemoryDataService implements InMemoryDbService {
   createDb() {
-    let workItems = [
+    let workitems = [
```
Comment (kwk) [inline src/app/in-memory-data.service.ts]: I'd leave this as `workItem`.

File: src/app/work-item/work-item.service.ts [hunk 2]
```diff
@@ -8,18 +9,19 @@ import { WorkItem } from './work-item';
...
+    logger.log("WorkItemService using url " + this.workItemUrl);
+  }
 
   getWorkItems(): Promise<WorkItem[]> {
     return this.http
       .get(this.workItemUrl)
       .toPromise()
-      // .then(response => response.json().data as WorkItem[])
-      .then(response => response.json() as WorkItem[])
+      .then(response => process.env.ENV!='inmemory'?response.json() as WorkItem[]:response.json().data as WorkItem[])
```
Comment (kwk) [inline src/app/work-item/work-item.service.ts]: This looks very packed but is not very readable.

**Comment by kwk** [comment]
Except for some spelling things this LGTM.

File: src/app/in-memory-data.service.ts (hunk 1 above)
Comment (michaelkleinhenz) [inline src/app/in-memory-data.service.ts]: I also would, but the name gets reflected as the url component and thus must be compatible with the "real" api.

File: src/app/work-item/work-item.service.ts (hunk 2 above)
Comment (michaelkleinhenz) [inline src/app/work-item/work-item.service.ts]: +1
But more readable than declaring an external func :-)

**Comment by joshuawilson** [comment]
I'm not sure how long we will keep the in-memory data set, but this is a nice addition to use it.

**Pull request merged**